Makefile

```diff
@@ -153,29 +158,73 @@ else
 KIND=$(shell which kind)
 endif
 
+# Prints relevant environment variables
+.PHONY: envs
+envs:
+	@{ \
+	echo "CONTROLLER_GEN=$(CONTROLLER_GEN)"; \
+	echo "KUSTOMIZE=$(KUSTOMIZE)"; \
+	echo "AUTHORINO_IMAGE=$(AUTHORINO_IMAGE)"; \
+	echo "AUTHORINO_NAMESPACE=$(AUTHORINO_NAMESPACE)"; \
+	echo "AUTHORINO_DEPLOYMENT=$(AUTHORINO_DEPLOYMENT)"; \
+	}
+
+# Creates a namespace where to deploy Authorino
+.PHONY: namespace
+namespace:
+	kubectl create namespace $(AUTHORINO_NAMESPACE)
+
+# Deploys the examples user apps: Talker API and Envoy proxy, and optionally Keycloak and Dex
+.PHONY: example-apps
+NAMESPACE ?= $(AUTHORINO_NAMESPACE)
+DEPLOY_KEYCLOAK ?= $(DEPLOY_IDPS)
+DEPLOY_DEX ?= $(DEPLOY_IDPS)
+example-apps:
+	kubectl -n $(NAMESPACE) apply -f examples/talker-api/talker-api-deploy.yaml
+	kubectl -n $(NAMESPACE) apply -f examples/envoy/envoy-deploy.yaml
+ifneq (, $(DEPLOY_KEYCLOAK))
+	kubectl -n $(NAMESPACE) apply -f examples/keycloak/keycloak-deploy.yaml
+endif
+ifneq (, $(DEPLOY_DEX))
+	kubectl -n $(NAMESPACE) apply -f examples/dex/dex-deploy.yaml
+endif
+
+# Targets with the 'local-' prefix, for trying Authorino in a local cluster spawned with Kind
+
 KIND_CLUSTER_NAME ?= authorino
 
 # Start a local Kubernetes cluster using Kind
 .PHONY: local-cluster-up
-local-cluster-up: kind local-cluster-down
+local-cluster-up: kind
 	kind create cluster --name $(KIND_CLUSTER_NAME) --config ./utils/kind-cluster.yaml
 
-# Deletes the local Kubernetes cluster started using Kind
-.PHONY: local-cluster-down
-local-cluster-down: kind
-	kind delete cluster --name $(KIND_CLUSTER_NAME)
-
-# Pushes a local container image of Authorino to the registry of the Kind-started local Kubernetes cluster
+# Pushes the Authorino image to the registry of the Kind-started local Kubernetes cluster
 .PHONY: local-push
 local-push: kind
-	kind load docker-image $(IMG) --name $(KIND_CLUSTER_NAME)
+	kind load docker-image $(AUTHORINO_IMAGE) --name $(KIND_CLUSTER_NAME)
+
+# Builds the image, pushes to the local cluster and deployes Authorino.
+# Sets the imagePullPolicy to 'IfNotPresent' so it doesn't try to pull the image again (just pushed into the server registry)
+.PHONY: deploy
+local-deploy: docker-build local-push deploy
+	kubectl -n $(AUTHORINO_NAMESPACE) patch deployment authorino-controller-manager -p '{"spec": {"template": {"spec":{"containers":[{"name": "manager", "imagePullPolicy":"IfNotPresent"}]}}}}'
```

Comment:
Has this been tried?
The issue with setting `ImagePullPolicy` to `IfNotPresent` is that if the cluster is already running the new image will not be used because the image will have the same tag as before.

Reply:
It has been tried and it works. This is because `local-deploy` builds the image and pushes it to the authorino kind cluster. As the image is tagged `latest` by default, with `ImagePullPolicy`, Kubernetes tries to pull the image, which obviously fails, despite the image being present within the cluster already. So we patch setting the policy to `IfNotPresent`.

Reply:
:+1: . I guess it fails because it tries to actually go to the repository. on the case of the default `Always` pullPolicy